src/python.c

```diff
@@ -1769,6 +1770,31 @@ canPushdownUpperrel(MulticornPlanState * state)
 		}
         Py_XDECREF(p_object);
 
+        /* Construct supported qual operators list */
+        p_ops = PyMapping_GetItemString(p_upperrel_pushdown, "operators_supported");
```

Comment:
Minor: this code looks very similar to the one that takes `agg_functions.keys()` and copies it into the `state->agg_functions` list just above; is it possible to refactor both into something like this?

```
void appendPySequenceToList(PyObject* p_object, List* target)
```

Reply:
Good catch, I'll add that function.